### PR TITLE
Add Minimum Window Substring problem, Hard, No 76

### PR DESCRIPTION
**Pull request by ipapast**

### CODE REVIEW

**Comment by ipapast** [comment]
Closes issue #553

**Comment by noisefilter19** [comment]
Unsure if this would work. Please follow the same format as in Leetcode.
They use class Solution{}

File: Java/MinimumWindowSubstring76.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
+import java.util.HashMap;
+import java.util.Map;
+
+class MinimumWindowSubstring {
```
Comment (noisefilter19) [inline Java/MinimumWindowSubstring76.java:4]: Different classname

File: Java/MinimumWindowSubstring76.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
+import java.util.HashMap;
```
Comment (noisefilter19) [inline Java/MinimumWindowSubstring76.java:1]: Add problem link and change the file name to have only the problem name with each word separated by _

Comment (ipapast) [reply Java/MinimumWindowSubstring76.java]: Done, please review if you are happy with the changes

**Comment by ipapast** [comment]
I made the changes as per your suggestions @noisefilter19 , let me know if you are not happy with them.

**Review by noisefilter19 — APPROVED** [review]

**Pull request merged**